### DISCUSSION TITLE
How to configure MQTT sensor topic

### DISCUSSION THREAD

**Discussion by ZonPV** [edterbak/HeishaMoNR#173] [Q&A] [answered]
In the past I was able to configure Solar2DHW on my own RPI and it runs well.

Now I want to install that functionality on the RPI of my brother too. He has also a Panasonic and Heishamon. Easy case I thought. 
Node-Red 4.0.5, MQTT with broker runs oke, Updated to flow version 24.04, His RPI runs on 192.168.2.20. P1-monitor, which send the MQTT messages, runs on a different RPI but I read the info on the RPI where all other software is running. Below a screenshot of MQTT Explorer on his RPI where also Node-Red runs.

![P1-monitor smartmeter 2-20](https://github.com/user-attachments/assets/fa60976b-6c22-4b46-834d-11bbde87d15a)

I activated Solar2DHW, In P1 setup, set Correction factor 1000.

Now the problem:
In System, Sensors I want to fill in 
-> ID 5 to p1monitor/smartmeter/production_kw
-> ID 6 to p1monitor/smartmeter/consumption_kw

but if I press with cursor on the pull down button there is no text.... The field stays empty, I cannot type there either.. 
No idea how to get further.

In MQTT Explorer I look at the broker on I see the topics I need. 
When I press Testbutton of Broker in System, MQTT it  says Connected and updates the time.

![MQTT ](https://github.com/user-attachments/assets/68973037-30b7-4581-8479-c4c26c732d16)

Any help is appreciated.

**Comment by edterbak** [comment, marked as answer]
![image](https://github.com/user-attachments/assets/9162f2c7-345f-4629-9c54-a12d6e5040ef)

When you have connected the link-in nodes from the correct sources, Save/Deploy it.
GO to the tab, system>sensors.
Press the **trashcans** behind both of the lines first.
they wait for the values to come in and then you can use the dropdown to select the source .

**Reply by ZonPV** [reply]
Sorry @edterbak , your answer was not directly clear to me.. 
But when I studied the working NR-flow on my NAS after you mentioned to look (again) at the WP Input flow than I saw an extra link In-node!! That was the missing trick!
So I added an extra flow-tab and exported from my working flow and imported to NR-flow on the RPI of my brother.
![Production](https://github.com/user-attachments/assets/774db179-0527-489d-979c-9adf61f99b79)

To be honest: what was my mistake as a beginner with NR? 
I selected the Comment node to look for properties to config them.. I had to select the above Link In-node..

Thnx for your quick hint!
![Link In node](https://github.com/user-attachments/assets/daf32da9-f45c-4f82-b61e-8ea9880088d6)

**Reply by edterbak** [reply]
How can I make it more clear to a user like yourself? what would help in this case :)
What would your suggestion be to improve it. 

Because I am in deep in NR and technical aware of things, I fear I sometimes mis the obvious difficulties a beginner user could struggle with. So your insights in this could help me and everyone else to :)

**Reply by ZonPV** [reply]
You wrote: 
_How can I make it more clear to a user like yourself? what would help in this case :)
What would your suggestion be to improve it._

You helped me earlier with this problem. I remembered that I asked this question before (in Tweakers-forum I guess), you answered and solved my problem. But I forgot the solution...

And I read the Help in the Dashboard but there is no info about how to configure the flow(s). So that would be an improvement. To be honest, I did not search in the Help of Github...

On the other hand: 
why is there need for a (small) flows in the Personal Tab and activate links in the WP Input flow? 
Maybe because the topics are not always the same in every application of P1-readers? No idea about that. 
But my suggestion is to implement them in the WP input flow. 
As I see now there can be several sources to deliver that data to the flow. 

By the way I did not disable the link out 83-node. As far as I can see it does not harm..

**Comment by edterbak** [comment]
You can correct the P1 multiplier settings also under System > sensor > P1 setup button. I moved it to a more logical location i version 24.04

**Reply by ZonPV** [reply]
I did that before.
Thnx for your flows and support!

**Reply by edterbak** [reply]
No worries !! Glad to be of service.
